### PR TITLE
chore: update migration guides for Analytics, InApp Messaging and Push Notifications

### PR DESCRIPTION
**Pull request by Samaritan1011001**
#### Description of changes:
Updates the migration guide for Analytics, InAppMessaging and Push Notifications sections to be more detailed intended to show differences in the APIs. 

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+## Record and Configuration APIs
+
+As of v6 of Amplify, you will now import the functional API’s directly from the `aws-amplify/analytics` path as shown below.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:34]:
```suggestion
As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/analytics` path as shown below. 
```

File: src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+## Initialization and syncing messages
+
+As of v6 of Amplify, you will now import the functional API’s directly from the `aws-amplify/in-app-messaging` path as shown below.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx:24]:
```suggestion
As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/in-app-messaging` path as shown below. 
```

File: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+## Initialize, request permissions, and receive device tokens
+
+As of v6 of Amplify, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or it's status and finally how to obtain the device token.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx:24]:
```suggestion
As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or it's status and finally how to obtain the device token. 
```

**Review by renebrandel — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
++     // OPTIONAL, the attributes of the event
++     attributes: {
++       customizableField: 'attr'
++     }
++   }
++ });
+```
+## Analytics.identifyUser
+
+You can identify an user with Analytics with `identifyUser` API,
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:89]:
```suggestion
You can identify an Analytics user with the `identifyUser` API, 
```

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+import { identifyUser } from 'aws-amplify/analytics';
+
+identifyUser({
+  userId: "<user-id>", // If the user was signed in through Amplify Auth's signIn method you can retrieve the current user's ID from there.
+  userProfile
+});
+
+```
+## Streaming data
+You can stream analytics data to Kinesis. To configure Kinesis with Amplify, reference the [Installation and Configuration](https://docs.amplify.aws/react/build-a-backend/more-features/analytics/streaming-data/#installation-and-configuration) section of the documentation. Below is an example on how to stream data and flush events as needed,
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:101]:
```suggestion
You can stream analytics data to Kinesis. To configure Kinesis with Amplify, reference the [Installation and Configuration](https://docs.amplify.aws/react/build-a-backend/more-features/analytics/streaming-data/#installation-and-configuration) section of the documentation. Below is an example on how to stream data and flush events,
```

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 6]
```diff
@@ -0,0 +1,140 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: '',
+  platforms: [
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:6]: As a sanity check, is this platform list correct? I haven't used the new doc site too much, but should this only be platforms that have a v5 to v6 migration guide for Analytics?

File: src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+- await InAppMessaging.clearMessages();
++ await clearMessages();
+
+```
+
+## IdentifyUser
+
+The input format of the `identifyUser` has slightly changed. The below table shows the differences in the input type.
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx:73]:
```suggestion
The input format of the `identifyUser` has changed slightly. The below table shows the differences in the input type.
```

File: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx [hunk 8]
```diff
@@ -0,0 +1,160 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: '',
+  platforms: ['react-native', 'flutter', 'swift', 'android']
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx:6]: Ditto with Swift & Android

File: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+## Initialize, request permissions, and receive device tokens
+
+As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or it's status and finally how to obtain the device token.
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx:24]:
```suggestion
As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or its status and finally how to obtain the device token. 
```

File: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
++   myNotificationOpenedHandler
++ );
+
+- await Notifications.Push.getLaunchNotification();
++ const launchNotification = await getLaunchNotification();
+
+```
+## IdentityUser
+
+Differences in identifying a user is as shown below. The input type change is identical to the changes made for the `identifyUser` of the In App Messaging category mentioned in its section.
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx:130]:
```suggestion
Differences in identifying a user is as shown below. The input type change is identical to the changes made for the `identifyUser` of the In App Messaging category mentioned above.
```

**Review by jimblanc — COMMENTED** [review]
Couple nits, otherwise looks pretty good

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx (hunk 6 above)
Comment (Samaritan1011001) [reply src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:6]: I kept the same list as the other pages under Analytics but I think what you are pointing out makes sense, we should have this for only the migration platforms. Updating the PR accordingly, thanks!

File: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx (hunk 8 above)
Comment (Samaritan1011001) [reply src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx:6]: TY. Only `react-native` should stay given there is migration needed only there.

Comment (Samaritan1011001) [reply src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx]: Updated it to just `['react-native']`

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx (hunk 6 above)
Comment (Samaritan1011001) [reply src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:6]: Updated it to, 
```
[
    'javascript',
    'react-native',
    'angular',
    'nextjs',
    'react',
    'vue'
  ]
```

**Review by jimblanc — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Review by sowmyadev-kota — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 11]
```diff
@@ -0,0 +1,137 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: '',
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:5]: Missing description for SEO

File: src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx [hunk 12]
```diff
@@ -0,0 +1,156 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: '',
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx:5]: Please add description for SEO

File: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx [hunk 13]
```diff
@@ -0,0 +1,160 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: '',
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx:5]: Please add description for SEO

File: src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx (hunk 12 above)
Comment (Samaritan1011001) [reply src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx]: Added, ty!

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 14]
```diff
@@ -0,0 +1,137 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: 'Learn more about the migration steps to upgrade to v6',
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:5]:
```suggestion
  description: 'Learn more about the migration steps to upgrade Analytics APIs for Amplify JavaScript v5 to v6',
```

File: src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx [hunk 15]
```diff
@@ -0,0 +1,156 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: 'Learn more about the migration steps to upgrade to v6',
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx:5]: description: 'Learn more about the migration steps to upgrade In-app messaging APIs for Amplify JavaScript v5 to v6',

File: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx [hunk 16]
```diff
@@ -0,0 +1,160 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: 'Learn more about the migration steps to upgrade to v6',
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx:5]: description: 'Learn more about the migration steps to upgrade Push Notifications APIs for Amplify JavaScript v5 to v6',

**Review by renebrandel — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Review by sowmyadev-kota — COMMENTED** [review]
LGTM

**Pull request merged**